In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb

In [2]:
df = pd.read_csv('train.csv')

#### Información general del set de datos

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7613 entries, 0 to 7612
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   id        7613 non-null   int64 
 1   keyword   7552 non-null   object
 2   location  5080 non-null   object
 3   text      7613 non-null   object
 4   target    7613 non-null   int64 
dtypes: int64(2), object(3)
memory usage: 297.5+ KB


#### Muestra aleatoria del set de datos

In [4]:
df.sample(5)

id    keyword         location  \
1964  2828    cyclone           Geneva   
2843  4088  displaced              NaN   
2500  3591   desolate       Pueblo, CO   
5944  8490   screamed  with Doflamingo   
3432  4908    explode         Winnipeg   

                                                   text  target  
1964  Back to the future in #Vanuatu how Cyclone Pam...       1  
2843  PennLive - Two families displaced by Mechanics...       1  
2500  @Bill_Roose That looks so desolate and just......       1  
5944  //kinda screamed &gt;_&lt; https://t.co/MSUY4q...       0  
3432       I'm ready to explode! http://t.co/OwJe3i6yGN       0

#### Descripción de variables estadísticas del set de datos

In [5]:
df.describe(include=[np.object, np.number]).T

count unique                                                top freq  \
id        7613    NaN                                                NaN  NaN   
keyword   7552    221                                         fatalities   45   
location  5080   3341                                                USA  104   
text      7613   7503  11-Year-Old Boy Charged With Manslaughter of T...   10   
target    7613    NaN                                                NaN  NaN   

             mean      std  min   25%   50%   75%    max  
id        5441.93  3137.12    1  2734  5408  8146  10873  
keyword       NaN      NaN  NaN   NaN   NaN   NaN    NaN  
location      NaN      NaN  NaN   NaN   NaN   NaN    NaN  
text          NaN      NaN  NaN   NaN   NaN   NaN    NaN  
target    0.42966  0.49506    0     0     0     1      1

#### Conversión de tipos de datos

- `id` toma valores del rango `[1, 10873]`, entonces puede usarse el tipo `uint16`.
- `target` puede ser `0` ó `1`.

In [6]:
df['id'] = df['id'].astype(np.uint16)
df['target'] = df['target'].astype(np.uint8)

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7613 entries, 0 to 7612
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   id        7613 non-null   uint16
 1   keyword   7552 non-null   object
 2   location  5080 non-null   object
 3   text      7613 non-null   object
 4   target    7613 non-null   uint8 
dtypes: object(3), uint16(1), uint8(1)
memory usage: 200.9+ KB
